packages/block-library/src/button/edit.js

```diff
@@ -133,10 +134,60 @@ function ButtonEdit( props ) {
 
 	const [ isEditingURL, setIsEditingURL ] = useState( false );
 	const isURLSet = !! url;
-	const opensInNewTab = linkTarget === NEW_TAB_TARGET;
-	const nofollow = !! rel?.includes( NOFOLLOW_REL );
+
 	const isLinkTag = 'a' === TagName;
 
+	// Defines how block attributes map to link value and vice versa.
+	const linkValueAttrsToDataMapping = {
```

Comment:
@bangank36 I've had a go at reimplementing your work on `getUpdatedLinkAttributes` but using these standardised functions and a mapping.

I'd welcome your testing and thoughts 🙇

Reply:
This looks promising! I'd to take a look and keep you posted